frontend/src/pages/notebookController/screens/server/StopServerModal.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import { Button, Modal, ModalVariant } from '@patternfly/react-core';
+import { useHistory } from 'react-router-dom';
+import { Notebook } from '../../../../types';
+import { deleteNotebook } from '../../../../services/notebookService';
+import useNamespaces from '../../useNamespaces';
+import useNotification from '../../../../utilities/useNotification';
+import { NotebookAdminContext } from '../admin/NotebookAdminContext';
+import { allSettledPromises } from '../../../../utilities/allSettledPromises';
+import { User } from '../admin/types';
+
+type StopServerModalProps = {
+  notebookToDelete?: Notebook | null;
+  setNotebookToDelete?: (notebook: Notebook | null) => void;
+};
+
+const StopServerModal: React.FC<StopServerModalProps> = ({
+  notebookToDelete,
+  setNotebookToDelete,
+}) => {
+  const history = useHistory();
+  const { notebookNamespace } = useNamespaces();
+  const notification = useNotification();
+  const [isModalShown, setModalShown] = React.useState<boolean>(false);
+  const [isDeleting, setDeleting] = React.useState(false);
+  const { dataToRefresh, setDataToRefresh } = React.useContext(NotebookAdminContext);
```

Comment:
This logic should have stayed in the Admin section, maybe in `NotebookAdminControl` (likely you need to create a new wrapper here, or take the provider up a level). You have made the `StopServerModel` know too much about where it is used -- which reduces its reusability. Ideally you want to generically handle items.